pkg/minikube/cluster/cluster.go

```diff
@@ -282,7 +283,19 @@ func (m *MachineConfig) CacheMinikubeISOFromURL() error {
 	}
 	defer out.Close()
 
-	if err = os.Rename(tmpISOFile.Name(), out.Name()); err != nil {
+	if err = renameFile(tmpISOFile, out); err != nil {
+		return nil
+	}
+
+	return nil
+}
+
+func renameFile(oldFile, newFile *os.File) error {
+	// File descriptor need to be closed otherwise it will throw error
```

Comment:
@praveenkumar At this moment the function is a private function, so limited to current package.  Also we are trying not to add more code to minikube package.  I am not sure what should be the right approach here. 
I think you are thinking that we should reuse the function in Minishift as it is easy to do the mistake what we we did last time.